### ISSUE TITLE
[BUG] Tasks API will no longer be able to store results of tasks in .tasks index after upgrade from ES 7.10 to OS 2.3

### ISSUE THREAD

**Issue by vikasvb90** [opensearch-project/OpenSearch#10240]
**Describe the bug**
An issue is observed where after upgrading cluster, Tasks API is no longer able to store results of tasks in .tasks index. This is due to change in mappings of .tasks index between versions ES 7.10 & OS 2.3. In OS 2.3 a new additional field cancelled[1] is added in the .tasks index. This seems to be bug which arises only with upgraded clusters. New clusters will not have any issue since .tasks index will be freshly created with correct mapping.

**To Reproduce**

1. Create an ElasticSearch 7.10 domain.
2. Store some task results in .tasks index. To do this use reindex operation as mentioned below. The below API call will return you a Task ID:
```
POST _reindex?wait_for_completion=false
{
  "source": {
    "index": "my-index-000001"
  },
  "dest": {
    "index": "my-new-index-000001"
  }
}
```

3. Now when you do GET _tasks/<task_id_retruned_from_above_apicall>, you will be able to successfully fetch the results of task.
4. Now, upgrade your domain to OS 2.3 and then try to repeat step 2 to store some task results.
5. Once you get the task ID, try to do GET _tasks/<task_id> and you will no longer be able to successfully get the results of task and you will be facing below error

```
{
  "error" : {
    "root_cause" : [
      {
        "type" : "resource_not_found_exception",
        "reason" : "task [ksqgLe2xSbyjfvZQMNfj_Q:152861] isn't running and hasn't stored its results"
      }
    ],
    "type" : "resource_not_found_exception",
    "reason" : "task [ksqgLe2xSbyjfvZQMNfj_Q:152861] isn't running and hasn't stored its results"
  },
  "status" : 404
}
```

This is because, task api is not able to store results in .tasks index due to mapping change in OS 2.3 version.

**Mappings of .tasks index in ES 7.10/OS 1.1/OS 1.2/OS 1.3:**
```
{
  ".tasks": {
    "mappings": {
      "_meta": {
        "version": 3
      },
      "dynamic": "strict",
      "properties": {
        "completed": {
          "type": "boolean"
        },
        "error": {
          "enabled": false,
          "type": "object"
        },
        "response": {
          "enabled": false,
          "type": "object"
        },
        "task": {
          "properties": {
            "action": {
              "type": "keyword"
            },
            "cancellable": {
              "type": "boolean"
            },
            "description": {
              "type": "text"
            },
            "headers": {
              "enabled": false,
              "type": "object"
            },
            "id": {
              "type": "long"
            },
            "node": {
              "type": "keyword"
            },
            "parent_task_id": {
              "type": "keyword"
            },
            "running_time_in_nanos": {
              "type": "long"
            },
            "start_time_in_millis": {
              "type": "long"
            },
            "status": {
              "enabled": false,
              "type": "object"
            },
            "type": {
              "type": "keyword"
            }
          }
        }
      }
    }
  }
}
```
**Mappings of .tasks index in OS 2.3:**
```
{
  ".tasks" : {
    "mappings" : {
      "dynamic" : "strict",
      "_meta" : {
        "version" : 3
      },
      "properties" : {
        "completed" : {
          "type" : "boolean"
        },
        "error" : {
          "type" : "object",
          "enabled" : false
        },
        "response" : {
          "type" : "object",
          "enabled" : false
        },
        "task" : {
          "properties" : {
            "action" : {
              "type" : "keyword"
            },
            "cancellable" : {
              "type" : "boolean"
            },
            "cancelled" : {
              "type" : "boolean"
            },
            "description" : {
              "type" : "text"
            },
            "headers" : {
              "type" : "object",
              "enabled" : false
            },
            "id" : {
              "type" : "long"
            },
            "node" : {
              "type" : "keyword"
            },
            "parent_task_id" : {
              "type" : "keyword"
            },
            "running_time_in_nanos" : {
              "type" : "long"
            },
            "start_time_in_millis" : {
              "type" : "long"
            },
            "status" : {
              "type" : "object",
              "enabled" : false
            },
            "type" : {
              "type" : "keyword"
            }
          }
        }
      }
    }
  }
}
```
**Workaround for now:**
Delete existing .tasks index and create new .tasks index with correct mapping. The API call to create .tasks index with correct mapping is below:

```
PUT .tasks
{
  "mappings": {
    "_meta": {
      "version": 3
    },
    "dynamic": "strict",
    "properties": {
      "completed": {
        "type": "boolean"
      },
      "error": {
        "enabled": false,
        "type": "object"
      },
      "response": {
        "enabled": false,
        "type": "object"
      },
      "task": {
        "properties": {
          "action": {
            "type": "keyword"
          },
          "cancellable": {
            "type": "boolean"
          },
          "cancelled": {
            "type": "boolean"
          },
          "description": {
            "type": "text"
          },
          "headers": {
            "enabled": false,
            "type": "object"
          },
          "id": {
            "type": "long"
          },
          "node": {
            "type": "keyword"
          },
          "parent_task_id": {
            "type": "keyword"
          },
          "running_time_in_nanos": {
            "type": "long"
          },
          "start_time_in_millis": {
            "type": "long"
          },
          "status": {
            "enabled": false,
            "type": "object"
          },
          "type": {
            "type": "keyword"
          }
        }
      }
    }
  },
  "settings": {
    "index": {
      "number_of_shards": "1",
      "auto_expand_replicas": "0-1",
      "number_of_replicas": "0"
    }
  }
}

```

**Expected Behavior**

* Task API should be able to store results in .tasks index even after upgrading domain from ES 7.10/OS 1.1/OS 1.2/OS 1.3 to OS 2.3
* We need to ensure that .tasks index is created with correct mappings after the upgrade so that tasks API would be able to store the results of tasks.

This bug occurred as part of this [commit](https://github.com/opensearch-project/OpenSearch/commit/22bfadf324ca4f9dafe2f65868f57b31ef25ab87).

**Comment by gaobinlong** [comment]
Unfortunately this bug was [fixed in 2.4](https://github.com/opensearch-project/OpenSearch/pull/5412), maybe you can try to upgrade you cluster to OS 2.4 or above.

**Comment by dblock** [comment]
Thanks @gaobinlong, I'm going to close this. @vikasvb90 reopen if you still see a problem in 2.4+.
